### PR TITLE
Add four bar linkage transmission

### LINKED ISSUE #91: Implement FourBarLinkageTransmission
http://docs.ros.org/melodic/api/transmission_interface/html/c++/classtransmission__interface_1_1FourBarLinkageTransmission.html

### PR DESCRIPTION
**Pull request by bmagyar**
Fix #91 
Porting FourBarLinkageTransmission from ROS1

### CODE REVIEW

**Review by jordan-palacios — COMMENTED** [review]
LGTM. Some small stuff.

File: transmission_interface/include/transmission_interface/accessor.hpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+template<typename T>
+std::vector<T> get_ordered_handles(
+  const std::vector<T> & unordered_handles, const std::vector<std::string> & names,
+  const std::string & interface_type)
+{
+  std::vector<T> result;
+  for (const auto & name : names) {
+    for (auto & handle : unordered_handles) {
+      if ((handle.get_name() == name) &&
+        (handle.get_interface_name() == interface_type) && handle)
```
[the remark this answers is not in the document]
Comment (bmagyar) [reply transmission_interface/include/transmission_interface/accessor.hpp:58]: there's operator bool for handles defined which checks for null

File: transmission_interface/include/transmission_interface/accessor.hpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+// limitations under the License.
+
+#ifndef TRANSMISSION_INTERFACE__ACCESSOR_HPP_
+#define TRANSMISSION_INTERFACE__ACCESSOR_HPP_
+
+#include <algorithm>
+#include <set>
+#include <string>
+#include <vector>
+
```
[the remark this answers is not in the document]
Comment (bmagyar) [reply transmission_interface/include/transmission_interface/accessor.hpp:22]: thanks, done!

File: transmission_interface/include/transmission_interface/accessor.hpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+template<class T>
+std::vector<std::string> get_names(const std::vector<T> & handles)
+{
+  std::set<std::string> names;
+  std::transform(
+    handles.cbegin(), handles.cend(), std::inserter(names, names.end()),
+    [](const auto & handle) {return handle.get_name();});
+  return std::vector<std::string>(names.begin(), names.end());
```
[the remark this answers is not in the document]
Comment (bmagyar) [reply transmission_interface/include/transmission_interface/accessor.hpp:46]: it's not happy with me when I try that

File: transmission_interface/include/transmission_interface/accessor.hpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+template<typename T>
+std::vector<T> get_ordered_handles(
+  const std::vector<T> & unordered_handles, const std::vector<std::string> & names,
+  const std::string & interface_type)
+{
+  std::vector<T> result;
+  for (const auto & name : names) {
+    for (auto & handle : unordered_handles) {
```
[the remark this answers is not in the document]
Comment (bmagyar) [reply transmission_interface/include/transmission_interface/accessor.hpp:56]: :+1: for more STL

File: transmission_interface/include/transmission_interface/accessor.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (jordan-palacios) [reply transmission_interface/include/transmission_interface/accessor.hpp:50]: Ah, the original `names` is a set, I didn't noticed that. Never mind then.

**Review by jordan-palacios — APPROVED** [review]

File: transmission_interface/include/transmission_interface/accessor.hpp [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  std::set<std::string> names;
+  std::transform(
+    handles.cbegin(), handles.cend(),
+    std::inserter(names, names.end()),
+    [](const auto & handle) {return handle.get_name();});
+  return std::vector<std::string>(names.begin(), names.end());
+}
+
+template<typename T>
+std::vector<T> get_ordered_handles(
```
Comment (destogl) [inline transmission_interface/include/transmission_interface/accessor.hpp:54]: We use this function all-around the library. Should we extract it in some helper-header?

**Review by destogl — APPROVED** [review]
a few nitpicks

File: transmission_interface/include/transmission_interface/accessor.hpp [hunk 6]
```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 PAL Robotics S.L.
```
[the remark this answers is not in the document]
Comment (bmagyar) [reply transmission_interface/include/transmission_interface/accessor.hpp:1]: Not very concerned with this TBH, and the changes are essentially from last year.

File: transmission_interface/include/transmission_interface/accessor.hpp (hunk 5 above)
Comment (bmagyar) [reply transmission_interface/include/transmission_interface/accessor.hpp:54]: Yes, that is exactly what this file is :) 
I'd refrain from creating too strong of a tie of code-base between `transmission_interface` and the others.

**Pull request merged**